File: Algoritmos e programação I/4. (27-03) Lista de Exercícios #1/Beecrowd_1045.py
'''
Beecrowd: Problema 1045
23/03/2024
Sophia Swiercoswski
'''

A, B, C = map(float, input().split())
lados = [A, B, C]
lados.sort(reverse=True)
A, B, C = lados

if A >= B + C:
    print ("NAO FORMA TRIANGULO")
elif (A * A) == (B * B) + (C * C):
    print ("TRIANGULO RETANGULO")
else:
    if (A * A) < (B * B) + (C * C):
        print ("TRIANGULO ACUTANGULO")
        if A == B == C:
            print ("TRIANGULO EQUILATERO")
        elif (A == B) or (B == C) or (A == C):
            print ("TRIANGULO ISOSCELES")
    elif (A * A) > (B * B) + (C * C):
        print ("TRIANGULO OBTUSANGULO")
        if (A == B) or (B == C) or (A == C):
            print ("TRIANGULO ISOSCELES")

File: Algoritmos e Programação I/C++/2. (21-08) Lista de Exercícios #4/Beecrowd_1045.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    // Criar array para armazenar 3 lados double e ordernar crescentemente
    double A, B, C, lados[3];
    for(int i = 0; i < 3; i++) {
        cin >> lados[i];
    }
    sort(lados, lados + 3);

    // Definir os lados, sendo A o maior
    A = lados[2];
    B = lados[1];
    C = lados[0];

    // Determinar o tipo de triângulo
    if(A >= B + C) {
        printf("NAO FORMA TRIANGULO\n");
    }
    else if((A * A) == (B * B) + (C * C)) {
        printf("TRIANGULO RETANGULO\n");
    }
    else {
        if((A * A) > (B * B) + (C * C)) {
            printf("TRIANGULO OBTUSANGULO\n");
        }
        else {
            printf("TRIANGULO ACUTANGULO\n");
        }

        if((A == B) && (B == C)) {
            printf("TRIANGULO EQUILATERO\n");
        }
        else if((A == B) || (B == C) || (C == A)) {
            printf("TRIANGULO ISOSCELES\n");
        }
    }

    return 0;
}